.github/workflows/upstream.yml

```
@@ -0,0 +1,130 @@
+      run:
+        shell: bash -l {0}
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@cb605e52c26070c328afc4562f0b4ada7618a84e # v2.10.4
+        with:
+          disable-sudo: true
+          egress-policy: block
+          allowed-endpoints: >
+            api.electricitymap.org:443
+            api.github.com:443
+            api.green-coding.io:443
+            api.securityscorecards.dev:443
+            conda.anaconda.org:443
+            dap.service.does.not.exist:443
+            files.pythonhosted.org:443
+            github.com:443
+            ip-api.com:80
+            ipapi.co:443
+            objects.githubusercontent.com:443
+            proxy.golang.org:4433
+            pypi.org:443
+            raw.githubusercontent.com:443
+            repo.anaconda.com:443
+            sum.golang.org:443
+      - name: Start Measurement
+        uses: green-coding-solutions/eco-ci-energy-estimation@e79d33b08b0c47de4438c161d432730fce23f246 # v4.2.0
+        with:
+          task: start-measurement
+          branch: ${{ github.head_ref || github.ref_name }}
+      - name: Checkout Repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          persist-credentials: false
+      - name: Setup Conda (Micromamba) with Python${{ matrix.python-version }}
+        uses: mamba-org/setup-micromamba@0dea6379afdaffa5d528b3d1dabc45da37f443fc # v2.0.4
+        with:
+          cache-downloads: true
+          cache-environment: true
+          environment-file: environment.yml
+          create-args: >-
+            pytest-reportlog
+            python=${{ matrix.python-version }}
+      - name: Install upstream versions
+        run: |
+          # git-based dependencies cannot be installed from hashes
+          python -m pip install -r CI/requirements_upstream.txt
+      - name: Install xsdba
+        run: |
+          python -m pip install --no-user --no-deps --editable .
+      - name: Check versions
+        run: |
+          micromamba list
+          xclim show_version_info
+          python -m pip check || true
+      - name: Setup Python Measurement
+        uses: green-coding-solutions/eco-ci-energy-estimation@e79d33b08b0c47de4438c161d432730fce23f246 # v4.2.0
+        with:
+          task: get-measurement
+          label: 'Environment Setup (Upstream, Python${{ matrix.python-version }})'
+        continue-on-error: true
+      - name: Test Data Caching
+        uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
+        with:
+          path: |
+            ${{ matrix.testdata-cache }}
+          key: ${{ runner.os }}-xclim-testdata-upstream-${{ hashFiles('pyproject.toml', 'tox.ini') }}
+      - name: Run Tests
+        if: success()
+        id: status
+        run: |
+          python -m pytest --numprocesses=logical --durations=10 --cov=xclim --cov-report=term-missing --report-log output-${{ matrix.python-version }}-log.jsonl
```

```suggestion
          python -m pytest --numprocesses=logical --durations=10 --cov=xclim --cov-report=term-missing --report-log output-${{ matrix.python-version }}-log.jsonl
```
--cov is coverage right? should this be xsdba or is this to cover the external library?

---

I'm glad that my attention was awarded with a useful correction lol, didn't expect I would find something